Userland/Libraries/LibJS/Runtime/BigIntConstructor.cpp

```diff
@@ -73,7 +73,14 @@ JS_DEFINE_NATIVE_FUNCTION(BigIntConstructor::as_int_n)
 // 21.2.2.2 BigInt.asUintN ( bits, bigint ), https://tc39.es/ecma262/#sec-bigint.asuintn
 JS_DEFINE_NATIVE_FUNCTION(BigIntConstructor::as_uint_n)
 {
-    TODO();
+    // 1. Set bits to ? ToIndex(bits).
+    auto bits = TRY(vm.argument(0).to_index(global_object));
+
+    // 2. Set bigint to ? ToBigInt(bigint).
+    auto* bigint = TRY(vm.argument(1).to_bigint(global_object));
+
+    // 3. Return the BigInt value that represents ℝ(bigint) modulo 2^bits.
+    return js_bigint(vm, modulo(bigint->big_integer(), "1"_bigint.shift_left(bits)));
```

Comment:
While this is technically correct (and could absolutely be merged as an initial implementation) the real solution here obviously is to truncate the big integer, since when you run `BigInt.asUintN(999999, 1);` on other engines the result is instantaneous :)